telegram/ext/filters.py

```diff
@@ -865,73 +865,83 @@ class user(BaseFilter):
+                self._usernames = {}
+            elif isinstance(username, str):
+                self._usernames = {username.replace('@', '')}
+            else:
+                self._usernames = set([user.replace('@', '') for user in username])
+
         @property
         def user_ids(self):
-            with self._user_ids_lock:
-                return self._user_ids
+            with self.__lock:
+                return frozenset(self._user_ids) if self._user_ids is not None else None
 
         @user_ids.setter
         def user_ids(self, user_id):
-            if (user_id is None) == (self.usernames is None):
-                raise ValueError('One and only one of user_id or username must be used')
-            with self._user_ids_lock:
-                if user_id is None:
-                    self._user_ids = None
-                elif isinstance(user_id, int):
-                    self._user_ids = set([user_id])
-                else:
-                    self._user_ids = set(user_id)
+            with self.__lock:
+                if user_id and self._usernames:
+                    raise RuntimeError(('Can\'t set user_id in conjunction with (already set) '
+                                        'usernames.'))
+                self._set_user_ids(user_id)
 
         @property
         def usernames(self):
-            with self._usernames_lock:
-                return self._usernames
+            with self.__lock:
+                return frozenset(self._usernames) if self._usernames is not None else None
 
         @usernames.setter
         def usernames(self, username):
-            if (username is None) == (self.user_ids is None):
-                raise ValueError('One and only one of user_id or username must be used')
-            with self._usernames_lock:
-                if username is None:
-                    self._usernames = None
-                elif isinstance(username, str):
-                    self._usernames = set([username.replace('@', '')])
-                else:
-                    self._usernames = set([user.replace('@', '') for user in username])
+            with self.__lock:
+                if username and self._user_ids:
+                    raise RuntimeError(('Can\'t set username in conjunction with (already set) '
+                                        'user_ids.'))
+                self._set_usernames(username)
 
         def filter(self, message):
             """"""  # remove method from docs
-            if self.user_ids is not None:
-                return bool(message.from_user and message.from_user.id in self.user_ids)
-            else:
-                # self.usernames is not None
-                return bool(message.from_user and message.from_user.username
-                            and message.from_user.username in self.usernames)
+            if bool(message.from_user):
+                if self.user_ids:
+                    return bool(message.from_user.id in self.user_ids)
```

Comment:
that's already a bool. you don't need to wrap it with another bool...

(and yes, i know it was like that before. doesn't meed we need to keep doing the same thing ;-) )